### PR TITLE
added exit status so failed tests exit with non-0 status

### PR DESCRIPTION
**Pull request by johncoleman83**
I added this to help me with my Continuous Integration, so my tests return non-0 exit status, thought you might want the addition too.

### CODE REVIEW

File: w3c_validator.py [hunk 1]
```diff
@@ -79,6 +79,7 @@ def __analyse_css(file_path):
 def __analyse(file_path):
     """Start analyse of a file and print the result
     """
+    RET_VAL = 0
```
Comment (papamuziko) [inline w3c_validator.py]: Please rename it to "nb_errors"

File: w3c_validator.py [hunk 2]
```diff
@@ -89,11 +90,21 @@ def __analyse(file_path):
...
         else:
             __print_stdout("{}: OK\n".format(file_path))
 
     except Exception as e:
         __print_stderr("[{}] {}\n".format(e.__class__.__name__, e))
+    return RET_VAL
+
+
+def __main_app():
+    RET_VAL = 0
```
Comment (papamuziko) [inline w3c_validator.py]: same

File: w3c_validator.py [hunk 3]
```diff
@@ -103,5 +114,4 @@ def __analyse(file_path):
         __print_stderr("usage: w3c_validator.py file1 file2 ...\n")
         exit(1)
 
-    for file_path in sys.argv[1:]:
-        __analyse(file_path)
+    sys.exit(__main_app())
```
Comment (papamuziko) [inline w3c_validator.py]: Please at a comment before this line and in the doctring on top of the file for explaining what the exit code will be = number of errors

**Comment by johncoleman83** [comment]
updated the code with renamed `nb_errors` variable and documentation for exit status

**Pull request merged**